Docs : crio-conf-5 formatting

petervandenabeele opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/kubernetes-sigs/cri-o/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**

* I tried to improve the crio.conf.5 markdown documentation
* I added a link to toml and crio.8 documentation
* ~I added newlines after the options as intended in the source document~
  I reverted this commit because it breaks man page rendering

**- How I did it**

* Adding markdown link references
* ~for the missing newlines, added a backslash at the end of some lines~
  (reverted)

**- How to verify it**

* validate the diff
* look at the rendered version at
https://github.com/petervandenabeele/cri-o/blob/docs-crio-conf-5-formatting/docs/crio.conf.5.md  
and compare it with the "previous" version at
https://github.com/kubernetes-sigs/cri-o/blob/master/docs/crio.conf.5.md

**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->
Fix links of crio.conf.5 documentation

TomSweeneyRedHat commented:
ok-to-test

TomSweeneyRedHat commented:
/ok-to-test

TomSweeneyRedHat commented:
@petervandenabeele LGTM, thanks very much for the contribution!

rhatdan commented:
@petervandenabeele How do these look as man pages?

petervandenabeele commented:
Hi @rhatdan thanks, that's an interesting question that I had not considered. I only looked at the markdown rendering, not really the roff rendering ... (while I do use and like man pages). 

I tried to use the `ronn` package (http://manpages.ubuntu.com/manpages/bionic/man1/ronn.1.html) to simulate how it would come out as man page in roff format and it's ok, but not "great" ...

My additions of the backslashes for the newlines are not properly understood (but not worse then before). I changed it to double newlines in the first 3 cases in an experiment below and that looks a bit better (and would also work in markdown, I think). Below is a part of the file, for review.

How are the man pages generated from these .md files?

```
root@26d26a33e4b0:/# ronn -m crio.conf.5.md > crio_simulated_man_page
root@26d26a33e4b0:/# cat crio_simulated_man_page 
CRIO(5)                                                                CRIO(5)



crio.conf  -  configuration file of the CRI-O OCI Kubernetes Container Runtime
daemonThe CRI-O configuration file specifies all of the  available  configura-
tion  options  and command-line flags for the crio(8) OCI Kubernetes Container
Runtime daemon ./crio.8.md, but in a TOML format that can be more easily modi-
fied and versioned.

       The default crio.conf is located at /etc/crio/crio.conf.The TOML format
       https://github.com/toml-lang/toml is used as the encoding of  the  con-
       figuration  file. Every option and subtable listed here is nested under
       a global "crio" table. No bare options are used. The format of TOML can
       be simplified to:



           [table]
           option = value

           [table.subtable1]
           option = value

           [table.subtable2]
           option = value



CRIO TABLE
       CRI-O  reads  its  storage defaults from the containers-storage.conf(5)
       file located at /etc/containers/storage.conf. Modify this storage  con-
       figuration  if you want to change the system's defaults. If you want to
       modify storage just for CRI-O, you can change the storage configuration
       options here.

       root="/var/lib/containers/storage"

       Path  to  the "root directory". CRI-O stores all of its data, including
       containers images, in this directory.

       runroot="/var/run/containers/storage"

       Path to the "run directory". CRI-O stores all  of  its  state  in  this
       directory.

       storage_driver="overlay"

       Storage  driver  used  to  manage the storage of images and containers.
       Please refer to containers-storage.conf(5) to see all available storage
       drivers.

       storage_option=[]\  List  to pass options to the storage driver. Please
       refer  to  containers-storage.conf(5)  to  see  all  available  storage
       options.

       file_locking=true\  If  set  to  false,  in-memory locking will be used
       instead of file-based locking.

       file_locking_path="/runc/crio.lock"\ Path to the lock file.

... [cut-off, enough material]
```

mrunalp commented:
We use go-md2man for the conversion.

giuseppe commented:
it looks like `go-md2man` doesn't honor the trailing `\`.

This is what I see in the generated man page:

```
CRIO TABLE
       CRI-O reads its storage defaults from the containers-storage.conf(5) file located at /etc/containers/storage.conf. Modify this storage configuration if you want to change the
       system's defaults. If you want to modify storage just for CRI-O, you can change the storage configuration options here.

       root="/var/lib/containers/storage"\
         Path to the "root directory". CRI-O stores all of its data, including containers images, in this directory.

       runroot="/var/run/containers/storage"\
         Path to the "run directory". CRI-O stores all of its state in this directory.

       storage_driver="overlay"\
         Storage driver used to manage the storage of images and containers. Please refer to containers-storage.conf(5) to see all available storage drivers.

       storage_option=[]\
         List to pass options to the storage driver. Please refer to containers-storage.conf(5) to see all available storage options.

       file_locking=true\
         If set to false, in-memory locking will be used instead of file-based locking.

       file_locking_path="/runc/crio.lock"\
         Path to the lock file.
```

petervandenabeele commented:
Hi @rhatdan is this good enough improvement for the Markdown case, or should I try other improvements? Thanks :-)

rhatdan commented:
@petervandenabeele Can you remove the \ and just put the lines on the same line?

petervandenabeele commented:
Thanks for the review :-) I have essentially reverted my commit where I added the back slashes.
Is that OK ?

giuseppe commented:
> Thanks for the review :-) I have essentially reverted my commit where I added the back slashes.
> Is that OK ?

could you just drop the commit?

You can use `git rebase -i origin/master` then delete the two lines for the commit that introduced the change and the revert.  You should then have just one line with the commit you want to keep.

petervandenabeele commented:
Thx @giuseppe . I have squashed the useful remaining code in 1 commit.
Since this was a force push anyway, I also rebased on upstream/master.
Now it should be a small, single commit Merge Request.

petervandenabeele commented:
/retest

mrunalp commented:
/test e2e_rhel

mrunalp commented:
/test e2e_fedora

mrunalp commented:
@rhatdan ptal.

rhatdan commented:
/lgtm

umohnani8 commented:
/approve
LGTM

petervandenabeele commented:
Ah, this was probably the correct way to correct the md5 formatting, instead of me trying to use back-slashes that break the man page rendering :

https://github.com/terraform-providers/terraform-provider-google/pull/3301/files#diff-0

@rhatdan
